### PR TITLE
Rename read/write to send/receive

### PR DESCRIPTION
**Pull request by rsoeldner**
.. as discussed #1596

### CODE REVIEW

File: server/src/main/scala/org/http4s/server/websocket/package.scala [hunk 1]
```diff
@@ -40,13 +40,13 @@ package object websocket {
     * @param status The status code to return to a client making a non-websocket HTTP request to this route
     */
   def WS[F[_]](
-      read: Stream[F, WebSocketFrame],
-      write: Sink[F, WebSocketFrame],
+      send: Stream[F, WebSocketFrame],
```
Comment (rsoeldner) [inline server/src/main/scala/org/http4s/server/websocket/package.scala]: @aeons I'm unsure how to handle the deprecation here

File: server/src/main/scala/org/http4s/server/websocket/package.scala [hunk 2]
```diff
@@ -17,8 +17,8 @@ package object websocket {
   /**
     * Build a response which will accept an HTTP websocket upgrade request and initiate a websocket connection using the
     * supplied exchange to process and respond to websocket messages.
-    * @param read     The read side of the Exchange represents the stream of messages that should be sent to the client
-    * @param write    The write side of the Exchange is a sink to which the framework will push the websocket messages
+    * @param send     The send side of the Exchange represents the stream of messages that should be sent to the client
+    * @param receive  The receive side of the Exchange is a sink to which the framework will push the websocket messages
```
Comment (cquiroz) [inline server/src/main/scala/org/http4s/server/websocket/package.scala]: Maybe the description of `receive` could be clearer on the line of `receiving messages from the client`

**Comment by cquiroz** [comment]
Looks good. I guess existing users will get a compilation error right away forcing them to change the code and it's a simple rename in this case

File: core/src/main/scala/org/http4s/websocket/Websocket.scala [hunk 3]
```diff
@@ -4,6 +4,9 @@ import fs2._
 import org.http4s.websocket.WebsocketBits.WebSocketFrame
 
 private[http4s] final case class Websocket[F[_]](
-    read: Stream[F, WebSocketFrame],
-    write: Sink[F, WebSocketFrame]
-)
+    @deprecatedName('read) send: Stream[F, WebSocketFrame],
```
Comment (ChristopherDavenport) [inline core/src/main/scala/org/http4s/websocket/Websocket.scala]: Can the deprecation be versioned here?

File: server/src/main/scala/org/http4s/server/websocket/package.scala (hunk 2 above)
Comment (rsoeldner) [reply server/src/main/scala/org/http4s/server/websocket/package.scala]: Maybe change `client` to `consumer` (on the send parameter) and `producer` on the `receive` parameter ?

File: server/src/main/scala/org/http4s/server/websocket/package.scala (hunk 1 above)
Comment (rossabaker) [reply server/src/main/scala/org/http4s/server/websocket/package.scala]: I don't think we can in this case.  It'll break for people using named arguments, but at least the types are unique.  I think it'll be okay.

File: core/src/main/scala/org/http4s/websocket/Websocket.scala [hunk 4]
```diff
@@ -4,6 +4,9 @@ import fs2._
 import org.http4s.websocket.WebsocketBits.WebSocketFrame
 
 private[http4s] final case class Websocket[F[_]](
-    read: Stream[F, WebSocketFrame],
-    write: Sink[F, WebSocketFrame]
-)
+    @deprecatedName('read, "0.18-M6") send: Stream[F, WebSocketFrame],
+    @deprecatedName('write, "0.18-M6") receive: Sink[F, WebSocketFrame]
```
Comment (rossabaker) [inline core/src/main/scala/org/http4s/websocket/Websocket.scala]: Sorry, one last nitpick: the version that will include this is 0.18.0-M7.

Comment (rsoeldner) [reply core/src/main/scala/org/http4s/websocket/Websocket.scala]: Whoop, right :+1:  Thank You

File: server/src/main/scala/org/http4s/server/websocket/package.scala (hunk 2 above)
Comment (jmcardon) [reply server/src/main/scala/org/http4s/server/websocket/package.scala]: Something like `The incoming messages to this service` works as well. In a sense, it's even easier to think about incoming and outgoing messages.

You could do that for the client as well. So you can actually say "The send side of the exchange represents the outgoing messages sent from this service."

This mostly following from @rossabaker 's incentive to make this agnostic to where its done, so it works with the client.

File: server/src/main/scala/org/http4s/server/websocket/package.scala (hunk 1 above)
Comment (jmcardon) [reply server/src/main/scala/org/http4s/server/websocket/package.scala]: :+1: to this as well. It should be a simple fix for users.

**Comment by rsoeldner** [comment]
@rossabaker next try :tada:

**Pull request merged**